packages/core/src/config.ts

```diff
@@ -693,9 +702,16 @@ const composeBundleConfig = (
             // user should use copy to keep origin file or use another separate entry to deal this
             let request = data.request;
             if (request[0] === '.') {
-              request = extname(request)
-                ? request.replace(/\.[^.]+$/, jsExtension)
-                : `${request}${jsExtension}`;
+              if (extname(request)) {
+                if (JS_EXTENSIONS_PATTERN.test(request)) {
+                  request = request.replace(/\.[^.]+$/, jsExtension);
+                } else {
+                  // If it does not match jsExtensionsPattern, we should do nothing, eg: ./foo.png
```

Comment:
What about `scss` or `less`

Reply:
It's about css redirect, sass / less / css modules shoul be externalized well in seperated PR I think.